src/shared/inspector/inspectorhotreloader.cpp

```diff
@@ -123,3 +126,39 @@ void InspectorHotreloader::resetAllFiles() {
   m_announced_files.clear();
   Navigator::instance()->reloadCurrentScreen();
 }
+
+void InspectorHotreloader::reloadWindow() {
+  auto engineHolder = QmlEngineHolder::instance();
+  QQmlApplicationEngine* engine =
+      static_cast<QQmlApplicationEngine*>(engineHolder->engine());
+  // Here is the main QML file.
+  if (!engineHolder->hasWindow()) {
+    logger.error() << "No Window to reload";
+    return;
+  }
+  logger.error() << "Closing and full reloading window";
+
+  int x = 0, y = 0;
+  // We may have multiple closed windows still in here.
+  // So best to go over them all.
```

Comment:
By saying we "may have multiple closed windows", do you mean lingering windows from other parts of the VPN client while clicking around?

Reply:
I mean from previous "reloads", as we will create a new Window and the last one may or may not have been deleted yet :)